File: src/main/java/io/openfuture/openmessanger/domain/Message.java
package io.openfuture.openmessanger.domain;

import java.time.ZonedDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Message {
    private String body;
    private String sender;
    private String receiver;
    private ZonedDateTime sentAt;
}

File: src/main/kotlin/io/openfuture/openmessenger/repository/entity/Message.kt
package io.openfuture.openmessenger.repository.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.LocalDateTime

@Table(name = "message")
@Entity
class Message {
    @Id
    var id: Int = 0
    var body: String? = null
    var sender: String? = null
    var recipient: String? = null
    var contentType: String? = null
    var receivedAt: LocalDateTime? = null
    var sentAt: LocalDateTime? = null
    var privateChatId: Int? = null
    var groupChatId: Int? = null
}